internal/support/logging/router_test.go

```diff
@@ -62,9 +66,17 @@ func (dp dummyPersist) reset() {
 func (dp *dummyPersist) closeSession() {
 }
 
+func mockDIC() *di.Container {
+	return di.NewContainer(di.ServiceConstructorMap{
+		bootstrapContainer.LoggingClientInterfaceName: func(get di.Get) interface{} {
+			return logger.NewMockClient()
+		},
+	})
+}
+
 func TestPing(t *testing.T) {
 	// create test server with handler
-	ts := httptest.NewServer(LoadRestRoutes())
+	ts := httptest.NewServer(LoadRestRoutes(mockDIC()))
```

Comment:
Can we make these tests more effective and to-the-point by simply targeting the REST handler. for example:

https://github.com/edgexfoundry/edgex-go/blob/8d215e236cccce0aabd5398d23074e7883ede7ba/internal/core/data/router_test.go#L185

This also means we can remove the function `func mockDIC()` and simply just use a `LoggingClient` mock to call the REST handler.

Reply:
@AnthonyMBonafide It gets a little challenging for the handlers which use `mux.Vars(r)`. To set the vars on the `*http.Request` directly relies on knowledge of Gorilla's internals, and knowing that the router is `gorilla/mux`.

I looked at the discussion in #2042 before doing this change since I had some questions about how to approach this. I couldn't find a test for a handler which used `mux.Vars(r)` that was refactored to being invoked directly. I spoke with @tsconn23 about this Monday and we agreed on this approach for the scope of this PR, specifically that a mock DI Container would be needed and `httptest.NewServer` would stay for now (@tsconn23, please correct me if I'm wrong).

If there's a handler which uses `mux.Vars(r)` that's had its tests refactored please let me know, I'll use that pattern.

Reply:
I absolutely don't want the DIC (real, mock, or otherwise) added to any of the tests.  I looked briefly at the code and see `mux.Vars()` is called in `getCriteria()`.  I'd recommend bubbling it up into the affected controller functions and passing the result of `mux.Vars()` into those affected functions.  This will allow you to pass in (and vary) the `map[string]string` value returned by `mux.Vars()` in the tests (and not have the tests take a dependency on the DIC).